src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1080,6 +1119,8 @@ public override int Execute(DocumentsOperationContext context)
                                         //the other side will receive negative ack and will retry sending again.
                                         document = new BlittableJsonReaderObject(_buffer + item.Position, item.DocumentSize, context);
                                         document.BlittableValidation();
+                                        var attachmentType = item.Flags.Contain(DocumentFlags.Revision) ? AttachmentType.Revision : AttachmentType.Document;                                        
```

Comment:
Pretty sure that we don't really care if this is revision or document.
We only care if the attachment hash exists

Reply:
The type effects the key we search for and it is importent, so we do care.

Reply:
Why do you need a type? All you need to do is check if the hash is there, and you already have this from the metadata.

Reply:
the type defines if the key of the attachment is constructed by the document id or it's change vector if it is a revision the change vector replace the id, why? I don't know I haven't written it, this is whats implemented.